pkg/generic/descriptor/http.go

```diff
@@ -32,25 +33,109 @@ const (
 	MIMEApplicationProtobuf = "application/x-protobuf"
 )
 
+var (
+	_ dhttp.RequestGetter  = &HTTPRequest{}
+	_ dhttp.ResponseSetter = &HTTPResponse{}
+)
+
 // HTTPRequest ...
 type HTTPRequest struct {
-	Header      http.Header
-	Query       url.Values
-	Cookies     Cookies
-	Method      string
-	Host        string
-	Path        string
 	Params      *Params // path params
+	Request     *http.Request
 	RawBody     []byte
 	Body        map[string]interface{}
 	GeneralBody interface{} // body of other representation, used with ContentType
 	ContentType MIMEType
+	bodyMap     *ast.Node
+}
+
+// GetHeader implements http.RequestGetter of dynamicgo
+func (req *HTTPRequest) GetHeader(key string) string {
+	return req.Request.Header.Get(key)
+}
+
+// GetCookie implements http.RequestGetter of dynamicgo
+func (req *HTTPRequest) GetCookie(key string) string {
+	if c, err := req.Request.Cookie(key); err == nil {
+		return c.Value
+	}
+	return ""
+}
+
+// GetQuery implements http.RequestGetter of dynamicgo
+func (req *HTTPRequest) GetQuery(key string) string {
```

Comment:
1. http.Request.URL is a pointer. Not sure whether it's possible to be nil
2. URL.Query() parses RawQuery (the string) every time, so if `GetQuery` is to be called multiple times for one Request, it may be slow.

Reply:
In the previous implementation, req.Query is `url.Values` which will only be parsed once.

Reply:
done